components/function-controller/design/extend_function_spec_proposal.md

```diff
@@ -0,0 +1,529 @@
+# Extending CRD spec for Serverless v1alpha2
+
+## Summary
+
+The current Serverless API allows for limited configuration of the generated Function's deployment. 
+Currently users can only use ENVs to pass 3rd party service credentials but it doesn't allow for volume mounted secrets (which becomes the industry standard for [service bindings](https://servicebinding.io/application-developer/)).
+Moreover users have no control over the annotations that are applied on the function runtime pod. This excludes functions pods from features enabled by annotations (i.e custom log parsers via `fluentbit.io/parser: my-regex-parser`).
+
+
+
+## Motivation
+
+Give Serverless users the ability to:
+- Configure volume mounted secrets for Function's subresources.
+- Configure labels and annotations for the Function runtime pod.
+
+### Goals
+
+- Add more flexibility to the Serverless API - enable volume mounted secrets and pod annotations.
+- Organise spec attributes belonging to runtime and build-time configuration (?)
+- Propose sample Function CR visualising different variants
+
+## Discussion points
+
+### Runtime, build-time separation
+
+Since Function CR is managing two workloads ( Deployment for runtime and Job for build-time ) we need to separate them in the spec in order to make it clear where do the mounts and annotations belong. 
```

Comment:
```suggestion
Since Function CR is managing two workloads (Deployment for runtime and Job for build-time) we need to separate them in the spec in order to make it clear where the mounts and annotations belong. 
```